lib/ip_list_write.go

```diff
@@ -161,54 +160,105 @@ func IPListWriteFromIPRangeStr(rStr string) error {
 
 // IPListWriteFromReader returns a list of IPs after reading from a reader; the
 // reader should have IPs per-line.
-func IPListWriteFromReader(r io.Reader) {
+func IPListWriteAllFromReader(r io.Reader) {
+	IPListWriteFromReader(r, true, true, true)
+}
+
+// IPListWriteFromReader returns a list of IPs after reading from a reader
+// from selected sources; the reader should have IPs per-line.
+func IPListWriteFromReader(
+	r io.Reader,
+	ip bool,
+	iprange bool,
+	cidr bool) {
 	scanner := bufio.NewScanner(r)
 	for scanner.Scan() {
 		ipStr := strings.TrimSpace(scanner.Text())
 		if ipStr == "" {
 			break
 		}
-
-		if err := IPListWriteFromIPRangeStr(ipStr); err == nil {
-			continue
-		}
-
-		if StrIsIPStr(ipStr) {
-			fmt.Println(ipStr)
-			continue
+		if iprange {
+			if err := IPListWriteFromIPRangeStr(ipStr); err == nil {
+				continue
+			}
 		}
-
-		if StrIsCIDRStr(ipStr) {
-			if err := IPListWriteFromCIDR(ipStr); err == nil {
+		if ip {
+			if StrIsIPStr(ipStr) {
+				fmt.Println(ipStr)
 				continue
 			}
 		}
+		if cidr {
+			if StrIsCIDRStr(ipStr) {
+				if err := IPListWriteFromCIDR(ipStr); err == nil {
+					continue
+				}
+			}
+		}
 
 		// simply ignore anything else.
 	}
 }
 
 // IPListWriteFromStdin returns a list of IPs from a stdin; the IPs should be 1
 // per line.
-func IPListWriteFromStdin() {
-	IPListWriteFromReader(os.Stdin)
+func IPListWriteAllFromStdin() {
+	IPListWriteAllFromReader(os.Stdin)
+}
+
+// IPListWriteFromStdin returns a list of IPs from a stdin from selected
+//sources; the IPs should be 1 per line.
+func IPListWriteFromStdin(
+	ip bool,
+	iprange bool,
+	cidr bool) {
+	IPListWriteFromReader(os.Stdin, ip, iprange, cidr)
 }
 
 // IPListWriteFromFile returns a list of IPs found in a file.
-func IPListWriteFromFile(pathToFile string) error {
+func IPListWriteAllFromFile(pathToFile string) error {
```

Comment:
This function's implementation should just be `return IPListWriteFromFIle(pathToFile, true, true, true)`.